pkg/operator/resource/resourceapply/generic.go

```diff
@@ -78,5 +80,22 @@ func ApplyDirectly(client kubernetes.Interface, manifests AssetFunc, files ...st
 		ret = append(ret, result)
 	}
 
+	if recorder != nil {
```

Comment:
Maybe provide a noop recorder to avoid nil checks?

Reply:
added NewFakeRecorder() that records events to a slice (handy for unit tests)...

Reply:
I don't think that's what Mo meant, NoopRecorder is different from FakeRecorder. The former can be used by folks who don't want to have events recorded, but ensures that one is always set and you don't have to perform this nil check, something like `ioutil.Discard`. The latter is **only** for tests.

Reply:
@soltysh @enj i don't think we should allow events to not be recorded at all. If a function is publishing events, they should be recorded and not ignored. This function is used in operators to apply changes to resources the operator manages. I think the point of events here is to always have timeline based on events for debugging of why the resource XYZ changed.

Reply:
That would still mean removing the nil check right?  As long as it is easy to get an event recorder, I can live without the noop (one can always create one if needed...).

Reply:
it is super easy ;-)

Reply:
With Michal we went as far as always requiring that recorder so that nil won't be necessary at all.